esp/src/src-react/components/DaliImport.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from "react";
+import { DefaultButton, DetailsList, DetailsListLayoutMode, IColumn } from "@fluentui/react";
+import { SizeMe } from "react-sizeme";
+import { csvParse } from "d3-dsv";
+import { DaliService } from "@hpcc-js/comms";
+import { scopedLogger } from "@hpcc-js/util";
+import { TableGroup } from "./forms/Groups";
+import { useConfirm } from "../hooks/confirm";
+import nlsHPCC from "src/nlsHPCC";
+import { HolyGrail } from "../layouts/HolyGrail";
+
+const logger = scopedLogger("src-react/components/DaliImport.tsx");
+
+const myDaliService = new DaliService({ baseUrl: "" });
+
+interface DaliImportProps {
+}
+
+export const DaliImport: React.FunctionComponent<DaliImportProps> = ({
+
+}) => {
+
+    const [columns, setColumns] = React.useState<IColumn[]>([]);
+    const [items, setItems] = React.useState<any[]>([]);
+    const [xml, setXml] = React.useState<string>("");
+    const [path, setPath] = React.useState<string>("");
+    const [add, setAdd] = React.useState(false);
+
+    const [DaliPromptConfirm, setDaliPromptConfirm] = useConfirm({
+        title: nlsHPCC.DaliAdmin,
+        message: nlsHPCC.DaliPromptConfirm,
+        onSubmit: React.useCallback(() => {
+            myDaliService.Import({ XML: xml, Path: path, Add: add }).then(response => {
+                const data = csvParse(response.Result);
```

Comment:
I am curious why the csvParse() is used in DaliImport and DaliDelete and the parser.parse() is used for DaliAdd and DaliCount. In DaliCount, the response.Result is an integer. In DaliAdd/DaliDelete/DaliImport, the response.Result is a string.

Reply:
For DaliCount using csvparse causes an error where WsDali.CountResponse.Result: number but WsDali.CountRequest.Path?: string... which is why we are using the parser.parse().

DaliAdd was switched back to csvParse, that was an error.